File: java_lib/src/main/java/wzq/com/lib/MList.java
package wzq.com.lib;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

/**
 * author:Created by WangZhiQiang on 2019/5/31.
 */
public class MList {

    public static void main(String[] args) {
        List<String> list = new ArrayList<String>();
        list.add("Hello");
        list.add("World");
        list.add("HAHAHAHA");
        //第一种遍历方法使用foreach遍历List
        for (String str : list) {            //也可以改写for(int i=0;i<list.size();i++)这种形式
            System.out.println(str);
        }

        //第二种遍历，把链表变为数组相关的内容进行遍历
        String[] strArray = new String[list.size()];
        list.toArray(strArray);
        for (int i = 0; i < strArray.length; i++) //这里也可以改写为  foreach(String str:strArray)这种形式
        {
            System.out.println(strArray[i]);
        }

        //第三种遍历 使用迭代器进行相关遍历

        Iterator<String> ite = list.iterator();
        while (ite.hasNext())//判断下一个元素之后有值
        {
            System.out.println(ite.next());
        }
    }
}

File: java_lib/src/main/java/wzq/com/lib/MList.kt
package wzq.com.lib

import java.util.*

/**
 * author:Created by WangZhiQiang on 2019/5/31.
 */
object MList {

    @JvmStatic
    fun main(args: Array<String>) {
        val list = ArrayList<String>()
        list.add("Hello")
        list.add("World")
        list.add("HAHAHAHA")
        //第一种遍历方法使用foreach遍历List
        for (str in list) {            //也可以改写for(int i=0;i<list.size();i++)这种形式
            println(str)
        }

        //第二种遍历，把链表变为数组相关的内容进行遍历
        val strArray = arrayOfNulls<String>(list.size)
        list.toTypedArray()
        for (i in strArray.indices)
        //这里也可以改写为  foreach(String str:strArray)这种形式
        {
            println(strArray[i])
        }

        //第三种遍历 使用迭代器进行相关遍历

        val ite = list.iterator()
        while (ite.hasNext())
        //判断下一个元素之后有值
        {
            println(ite.next())
        }
    }
}
